### PR TITLE
tests: change Travis to use Docker

### PR DESCRIPTION
**Pull request by ntarocco**

### CODE REVIEW

**Comment by ntarocco** [comment]
@ppanero I was thinking to use `docker-services-cli`, but I am not sure it was easy to test multiple versions... Maybe we can add it... anyway I have copied the `docker-services.yml` from there.

**Comment by ntarocco** [comment]
@lnielsen to use Postgres 12, we need to upgrade SQLAlchemy min version from 1.1 to 1.2.18

File: .travis.yml [hunk 1]
```diff
@@ -51,32 +48,42 @@ matrix:
...
 
 install:
   - "travis_retry pip install -r .travis-${REQUIREMENTS}-requirements.txt"
   - "travis_retry pip install -e .[docs,versioning,tests,$EXTRAS]"
-  - "psql --version"
+
+before_script:
+  - "docker --version"
+  - "docker-compose --version"
+  - if [ -n "${DOCKER_SERVICE}" ]; then
```
Comment (ppanero) [inline .travis.yml:78]: This should not be needed: Which case triggers this if? I am not able to understand it.

File: .travis.yml [hunk 2]
```diff
@@ -51,32 +48,42 @@ matrix:
...
+    docker-compose -f docker-services.yml up -d "${DOCKER_SERVICE}"; fi
 
 script:
   - "./run-tests.sh"
 
+after_script:
+  - docker-compose -f docker-services.yml down
```
Comment (ppanero) [inline .travis.yml:85]: Should not be needed either. Should use `docker-services-cli down`

File: .travis.yml [hunk 3]
```diff
@@ -16,31 +16,28 @@ language: python
...
-
 env:
+  global:
+    - DOCKER_COMPOSE_VERSION=1.26.2
   matrix:
     - EXTRAS=sqlite REQUIREMENTS=lowest SQLALCHEMY_DATABASE_URI="sqlite:///test.db"
-    - EXTRAS=mysql REQUIREMENTS=lowest SQLALCHEMY_DATABASE_URI="mysql+pymysql://travis@localhost:3306/invenio"
-    - EXTRAS=postgresql REQUIREMENTS=lowest SQLALCHEMY_DATABASE_URI="postgresql+psycopg2://postgres@localhost:5432/invenio" POSTGRESQL_VERSION="9.6"
-    - EXTRAS=postgresql REQUIREMENTS=lowest SQLALCHEMY_DATABASE_URI="postgresql+psycopg2://postgres@localhost:5432/invenio" POSTGRESQL_VERSION="10"
+    - EXTRAS=mysql REQUIREMENTS=lowest DOCKER_SERVICE=mysql SQLALCHEMY_DATABASE_URI="mysql+pymysql://testuser:testpsw@localhost:3306/invenio" MYSQL_VERSION="5.7"
```
Comment (ppanero) [inline .travis.yml:24]: We should be able to avoid all the `DOCKER_SERVICE` and `URIs`.

File: docker-services.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
+# -*- coding: utf-8 -*-
```
Comment (ppanero) [inline docker-services.yml:1]: This file should not be needed.

**Review by ppanero — COMMENTED** [review]
Okay, I made some comments and then I realized actually the problem was that `docker-compose-cli` was not easy to integrate. So most likely you can ignore my comments 😅 . Could we have a chat IRL so I fully understand the issues?  

If there is no rush to integrate this I would like to try to integrate `docker-services-cli` and potentially fix the problems. Maybe is just a matter of improving the documentation (sorry about that).

*Update*: Made a PR with some required changes in the cli part https://github.com/inveniosoftware/docker-services-cli/pull/10.

**Comment by lnielsen** [comment]
> @lnielsen to use Postgres 12, we need to upgrade SQLAlchemy min version from 1.1 to 1.2.18

Ok from my side.

**Comment by ntarocco** [comment]
closed in favor of: https://github.com/inveniosoftware/invenio-db/pull/134

**Pull request closed without merging**